pkg/cli/cli.go

```diff
@@ -0,0 +1,276 @@
+	return &cobra.Command{
+		Use:   c.commandName,
+		Short: "Development kit for building Kubernetes extensions and tools.",
+		Long: fmt.Sprintf(`Development kit for building Kubernetes extensions and tools.
+
+Provides libraries and tools to create new projects, APIs and controllers.
+Includes tools for packaging artifacts into an installer container.
+
+Typical project lifecycle:
+
+- initialize a project:
+
+  %s init --domain example.com --license apache2 --owner "The Kubernetes authors"
+
+- create one or more a new resource APIs and add your code to them:
+
+  %s create api --group <group> --version <version> --kind <Kind>
+
+Create resource will prompt the user for if it should scaffold the Resource and / or Controller. To only
+scaffold a Controller for an existing Resource, select "n" for Resource. To only define
+the schema for a Resource without writing a Controller, select "n" for Controller.
+
+After the scaffold is written, api will run make on the project.
+`, c.commandName, c.commandName),
+		Example: fmt.Sprintf(`
+  # Initialize your project
+  %s init --domain example.com --license apache2 --owner "The Kubernetes authors"
+
+  # Create a frigates API with Group: ship, Version: v1beta1 and Kind: Frigate
+  %s create api --group ship --version v1beta1 --kind Frigate
+
+  # Edit the API Scheme
+  nano api/v1beta1/frigate_types.go
+
+  # Edit the Controller
+  nano controllers/frigate_controller.go
+
+  # Install CRDs into the Kubernetes cluster using kubectl apply
+  make install
+
+  # Regenerate code and run against the Kubernetes cluster configured by ~/.kube/config
+  make run
+`, c.commandName, c.commandName),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+		},
+	}
+}
+
+// cmdErr updates a cobra command to output error information when executed
+// or used with the help flag.
+func cmdErr(cmd *cobra.Command, err error) {
+	cmd.Long = fmt.Sprintf("%s\nNote: %v", cmd.Long, err)
+	cmd.RunE = errCmdFunc(err)
+}
+
+// errCmdFunc returns a cobra RunE function that returns the provided error
+func errCmdFunc(err error) func(*cobra.Command, []string) error {
+	return func(*cobra.Command, []string) error {
+		return err
+	}
+}
+
+// getProjectVersion tries to load PROJECT file and returns if the file exist
+// and the version string
+func getProjectVersion() (bool, string) {
+	if _, err := os.Stat("PROJECT"); os.IsNotExist(err) {
+		return false, ""
+	}
+	projectInfo, err := scaffold.LoadProjectFile("PROJECT")
+	if err != nil {
+		log.Fatalf("failed to read the PROJECT file: %v", err)
+	}
+	return true, projectInfo.Version
+}
+
+// getBaseFlags parses the command line arguments, looking for --project-version
+// and help. If an error occurs or only --help is set, getBaseFlags returns an
+// empty string and true. Otherwise, getBaseFlags returns the project version
+// and false.
+func (c CLI) getBaseFlags() (string, bool) {
+	fs := pflag.NewFlagSet("base", pflag.ExitOnError)
```

Comment:
FYI, this logic yields "unknown flag: project-version" when we get to the main flag parsing logic.  We prob want this defined on the main command too.